src/app/modules/products/pages/product/product.page.ts

```diff
@@ -12,27 +19,90 @@ import { Product } from "@shared/models/classes/product.class";
 })
 export class ProductPageComponent implements OnInit, OnDestroy {
     private subscriptions = new Subscription();
+    private products: BehaviorSubject<Product[]> = new BehaviorSubject<Product[]>([]);
     featuredProducts$: Observable<Product[]> = new Observable();
-    products$: Observable<Product[]> = new Observable();
+    filterableProperties: FilterType[] | undefined;
+    activeFilters: ActiveFilters | undefined;
+    filterForm = this.fb.group({});
+
+    get products$(): Observable<Product[]> {
+        return this.products.asObservable();
+    }
 
     constructor(
         private productService: ProductApiService,
         private route: ActivatedRoute,
+        private fb: FormBuilder,
     ) {}
 
     ngOnInit(): void {
         this.featuredProducts$ = this.productService.getFeaturedProducts();
 
-        this.subscriptions.add(this.route.params
-            .subscribe(
-                ({productType}) => {
-                    this.products$ = this.productService.getProductsByType(productType);
-                }
+        this.subscriptions.add(
+            this.route.params.subscribe(
+                ({productType}) =>
+                    this.subscriptions.add(
+                        this.productService.getProductsByType(productType).subscribe(
+                            products => {
+                                this.products.next(products);
+                                this.updateFilterForm(products, productType);
+                            })
+                    )
+            )
+        );
+
+        this.subscriptions.add(
+            this.filterForm.valueChanges.subscribe(
+                changes => this.updateActiveFilters(changes)
             )
         );
     }
 
     ngOnDestroy(): void {
         this.subscriptions.unsubscribe();
     }
+
+    private updateFilterForm(products: Product[], productType: ProductType): void {
+        this.filterableProperties
```

Comment:
rather than looping through all filterableProperties. We should extract the new filterable properties, check if the key exists and remove/add the existing keys